test-app/tests/helpers/utility-methods.js

```diff
@@ -20,139 +24,144 @@ export function fetchJSON({ url, params, method = 'GET' } = {}) {
+        attrs = {
+          owner: { key: 'humanId' },
+        };
+      };
+    case 'cat':
+      return class CatSerializer extends SerializerClass {
+        primaryKey = 'catId';
+        attrs = {
+          name: { key: 'catName' },
+          friend: 'catFriend',
+        };
+      };
+    case 'comic-book':
+      return class ComicBookSerializer extends SerializerClass.extend(
+        EmbeddedRecordsMixin,
+      ) {
+        attrs = {
+          includedVillains: { embedded: 'always' },
+          company: { embedded: 'always' },
+          characters: { embedded: 'always' },
+        };
+      };
+    default:
+      return SerializerClass;
   }
-  modelSerializer.store = store;
-  return modelSerializer;
 }
 
 // serializerType like -rest or -active-model, -json-api, -json
 export function containerSetup(application, serializerType) {
-  // brute force setting the adapter/serializer on the store.
-  if (serializerType) {
-    application.register('adapter:-active-model', ActiveModelAdapter, {
-      singleton: false,
-    });
-    application.register('serializer:-active-model', ActiveModelSerializer, {
-      singleton: false,
-    });
-
-    let store = application.lookup('service:store');
-
-    let adapterType = serializerType === '-json' ? '-rest' : serializerType;
-    let adapter;
-    if (serializerType === '-json' || serializerType === '-rest') {
-      application.register(`adapter:-rest`, RESTAdapter, {
-        singleton: false,
-      });
-    }
-    adapter = application.lookup('adapter:' + adapterType);
-    adapter = adapter.reopen();
+  assert(
+    'unexpected serializerType for test setup',
+    ['-rest', '-active-model', '-json-api', '-json'].includes(serializerType),
+  );
 
-    serializerType = serializerType === '-json' ? '-default' : serializerType;
-    let serializer = application.lookup('serializer:' + serializerType);
+  const modelNames = [
+    'big-group',
+    'big-hat',
+    'cat',
+    'comic-book',
+    'company',
+    'cool-stoner',
+    'dog',
+    'employee',
+    'entry-type',
+    'entry',
+    'fluffy-material',
+    'group',
+    'hat',
+    'manager',
+    'material',
+    'outfit',
+    'person',
+    'philosopher',
+    'profile',
+    'project',
+    'property',
+    'review',
+    'rod',
+    'salary',
+    'small-company',
+    'small-group',
+    'small-hat',
+    'soft-material',
+    'stoner',
+    'super-hero',
+    'user',
+    'villain',
+  ];
 
-    store.adapterFor = () => adapter;
+  const { adapter, serializer } = getBaseFor(serializerType);
+  application.register(`adapter:application`, adapter);
+  application.register(`serializer:application`, serializer);
+  application.register(`adapter:${serializerType}`, adapter);
+  application.register(`serializer:${serializerType}`, serializer);
 
-    let findSerializer = store.serializerFor.bind(store);
+  modelNames.forEach((modelName) => {
+    application.register(`adapter:${modelName}`, adapter);
```

Comment:
for each model, register the standard adapter of the type requested for this test module